### DISCUSSION TITLE
Tweaking LOG to be more UVM-like and cleaning up

### DISCUSSION THREAD

**Discussion by svenka3** [pyuvm/pyuvm#126] [Ideas]
I understand pyuvm doesn't want to implement UVM's native logging instead prefers Python/Cocotb logging scheme (atleast for now). That's a fair decision, I respect that. I have 2 inputs:

1. I see the "full_name" sort of getting repeated for UVM-like messages in pyuvm. An example below:

```log
500000.00ns INFO     ..op.env.scoreboard139829378200912 testbench.py(215)[uvm_test_top.env.scoreboard]: PASSED: 0xfe ADD 0x70 = 0x016e

```
I see that PyuvmFormatter class adds file/line and full_name and it works well. However, the SimColourLogFormatter from Cocotb (and the ones under that class) attempt to add "record.name" which in UVM Component cases happens to be same as UVM's full_name (I hope). 

So I propose that we get rid of the duplication in this case. Also due to some strange reason (didn't debug too much), the record.name has some funny extended numbers at the end such as .**.op.env.scoreboard_139829378200912_**

The second, minor perhaps for some, key for UVM fans is the "UVM_INFO/UVM_WARNING/UVM_ERROR/UVM_FATAL" as severity keywords in log file. I see that pyuvm leverages on built-in logging - but with a small tweak I managed to add a prefix to the INFO/WARNING/ERROR string. Below are samples:

Original pyvum log:
```log
                                                        Test ALU random and max forked
500000.00ns INFO     ..op.env.scoreboard139829378200912 testbench.py(215)[uvm_test_top.env.scoreboard]: PASSED: 0xfe ADD 0x70 = 0x016e
500000.00ns INFO     ..op.env.scoreboard139829378200912 testbench.py(215)[uvm_test_top.env.scoreboard]: PASSED: 0xff ADD 0xff = 0x01fe

```
My local, sandbox build's log:
```log
500000.00ns UVM_INFO        testbench.py(215)[uvm_test_top.env.scoreboard]: PASSED: 0xfb ADD 0x59 = 0x0154
500000.00ns UVM_INFO        testbench.py(215)[uvm_test_top.env.scoreboard]: PASSED: 0xff ADD 0xff = 0x01fe

```
Now, do you folks like this change? I will need to clean up and show my tweaks before I can submit PR, so your inputs would be valuable. 

Cheers
Srini

**Comment by raysalemi** [comment]
It appears these changes were made to **cocotb** rather than **pyuvm**.  Is that correct?

**Reply by svenka3** [reply]
Technically speaking yes, but practically - I extended logging.formatter class inside pyuvm s06...py file and used that extended class as base for pyuvmformatter 

I will be glad to show the code (bit ugly, will need to cleanup a bit) via a local branch if you like the idea.

Thanks

**Reply by svenka3** [reply]
https://github.com/svenka3/pyuvm/blob/master/pyuvm/tmp_s06_uvm_log.py

Kindly review the above file (it is modified on top of s06..py file. I will create a PR with same named file later if you folks like this approach).

**Reply by raysalemi** [reply]
I like the approach to getting rid of the extra text, but I don't want to use `UVM_*` in the label as I think this threatens to confuse people to think that the UVM reporting system has been implemented.  Also, this could break many downstream log readers.  Finally, it does not match the examples in the book.

If I read your code correct, your approach does *not* use the `UVM_*` preface, correct?
